Libraries/LibWeb/Fetch/Infrastructure/HTTP/Bodies.cpp

```diff
@@ -71,37 +72,31 @@ void Body::fully_read(JS::Realm& realm, Web::Fetch::Infrastructure::Body::Proces
     auto task_destination_object = task_destination.get<GC::Ref<JS::Object>>();
 
     // 2. Let successSteps given a byte sequence bytes be to queue a fetch task to run processBody given bytes, with taskDestination.
-    auto success_steps = [&realm, process_body, task_destination_object = task_destination_object](ReadonlyBytes bytes) -> ErrorOr<void> {
-        // Make a copy of the bytes, as the source of the bytes may disappear between the time the task is queued and executed.
-        auto bytes_copy = TRY(ByteBuffer::copy(bytes));
-        queue_fetch_task(*task_destination_object, GC::create_function(realm.heap(), [process_body, bytes_copy = move(bytes_copy)]() mutable {
-            process_body->function()(move(bytes_copy));
+    auto success_steps = [&realm, process_body, task_destination_object](ByteBuffer bytes) -> void {
+        queue_fetch_task(*task_destination_object, GC::create_function(realm.heap(), [process_body, bytes = move(bytes)]() mutable {
+            process_body->function()(move(bytes));
         }));
-        return {};
     };
 
     // 3. Let errorSteps optionally given an exception exception be to queue a fetch task to run processBodyError given exception, with taskDestination.
-    auto error_steps = [&realm, process_body_error, task_destination_object](GC::Ptr<WebIDL::DOMException> exception) {
-        queue_fetch_task(*task_destination_object, GC::create_function(realm.heap(), [process_body_error, exception]() {
-            process_body_error->function()(exception);
+    auto error_steps = [&realm, process_body_error, task_destination_object](JS::Value error) {
+        queue_fetch_task(*task_destination_object, GC::create_function(realm.heap(), [process_body_error, error]() {
+            process_body_error->function()(error);
         }));
     };
 
     // 4. Let reader be the result of getting a reader for body’s stream. If that threw an exception, then run errorSteps with that exception and return.
+    HTML::TemporaryExecutionContext const execution_context { realm, HTML::TemporaryExecutionContext::CallbacksEnabled::Yes };
+    auto reader_or_exception = acquire_readable_stream_default_reader(m_stream);
+    if (reader_or_exception.is_exception()) {
+        auto throw_completion = Bindings::dom_exception_to_throw_completion(realm.vm(), reader_or_exception.release_error());
+        error_steps(*throw_completion.value());
+        return;
+    }
+    auto reader = reader_or_exception.release_value();
+
     // 5. Read all bytes from reader, given successSteps and errorSteps.
-    // FIXME: Use streams for these steps.
```

Comment:
FYI, see: https://github.com/LadybirdBrowser/ladybird/pull/2162#issuecomment-2521178568 which is why I think this hasn't been done before

Reply:
If we don't apply this patch because of that though, we should at the very least say that in the comment here 😅

Reply:
Dang it! And here I thought I was doing something original :P. This patch does end up making YouTube load a lot better. respects the system theme even.

Reply:
Do either you or @trflynn89 feel keyed-in enough to this issue to discuss it with the spec folks directly? We can ask for it to be included in a WHATNOT call (or even schedule a bespoke call with more fetch-y experts)

Reply:
I haven't spent enough time investigating this one myself personally to understand it fully yet for that, though I do keep meaning to dig into it along with some other streams stuff that needs fixing when I find the time.

Reply:
Commit removed, now this is a branch about the HTMLImageElement problem